src/equations/compressible_euler_quasi_1d.jl

```diff
@@ -0,0 +1,379 @@
+as defined in [`initial_condition_convergence_test`](@ref).
+"""
+@inline function source_terms_convergence_test(u, x, t,
+                                               equations::CompressibleEulerEquationsQuasi1D)
+    # Same settings as in `initial_condition_convergence_test`. 
+    #Derivatives calculated with ForwardDiff.jl
+    c = 2
+    A = 0.1
+    L = 2
+    f = 1 / L
+    ω = 2 * pi * f
+    x1, = x
+    ini(x1, t) = c + A * sin(ω * (x1 - t))
+
+    rho(x1, t) = ini(x1, t)
+    v1(x1, t) = 1.0
+    e(x1, t) = ini(x1, t)^2 / rho(x1, t)
+    p1(x1, t) = (equations.gamma - 1) * (e(x1, t) - 0.5 * rho(x1, t) * v1(x1, t)^2)
+    a(x1, t) = 1.5 - 0.5 * cos(x1 * pi)
+
+    arho(x1, t) = a(x1, t) * rho(x1, t)
+    arhou(x1, t) = arho(x1, t) * v1(x1, t)
+    aE(x1, t) = a(x1, t) * e(x1, t)
+
+    darho_dt(x1, t) = ForwardDiff.derivative(t -> arho(x1, t), t)
+    darhou_dx(x1, t) = ForwardDiff.derivative(x1 -> arhou(x1, t), x1)
+
+    arhouu(x1, t) = arhou(x1, t) * v1(x1, t)
+    darhou_dt(x1, t) = ForwardDiff.derivative(t -> arhou(x1, t), t)
+    darhouu_dx(x1, t) = ForwardDiff.derivative(x1 -> arhouu(x1, t), x1)
+    dp1_dx(x1, t) = ForwardDiff.derivative(x1 -> p1(x1, t), x1)
+
+    auEp(x1, t) = a(x1, t) * v1(x1, t) * (e(x1, t) + p1(x1, t))
+    daE_dt(x1, t) = ForwardDiff.derivative(t -> aE(x1, t), t)
+    dauEp_dx(x1, t) = ForwardDiff.derivative(x1 -> auEp(x1, t), x1)
+
+    du1 = darho_dt(x1, t) + darhou_dx(x1, t)
+    du2 = darhou_dt(x1, t) + darhouu_dx(x1, t) + a(x1, t) * dp1_dx(x1, t)
+    du3 = daE_dt(x1, t) + dauEp_dx(x1, t)
+
+    return SVector(du1, du2, du3, 0.0)
+end
+
+"""
+    boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
+                                 surface_flux_function, equations::CompressibleEulerEquationsQuasi1D)
+Determine the boundary numerical surface flux for a slip wall condition.
+Imposes a zero normal velocity at the wall.
+Density is taken from the internal solution state and pressure is computed as an
+exact solution of a 1D Riemann problem. Further details about this boundary state
+are available in the paper:
+- J. J. W. van der Vegt and H. van der Ven (2002)
+  Slip flow boundary conditions in discontinuous Galerkin discretizations of
+  the Euler equations of gas dynamics
+  [PDF](https://reports.nlr.nl/bitstream/handle/10921/692/TP-2002-300.pdf?sequence=1)
+
+  Should be used together with [`TreeMesh`](@ref).
+"""
+@inline function boundary_condition_slip_wall(u_inner, orientation,
+                                              direction, x, t,
+                                              surface_flux_function,
+                                              equations::CompressibleEulerEquationsQuasi1D)
+    # compute the primitive variables
+    rho_local, v_normal, p_local, a_local = cons2prim(u_inner, equations)
+
+    if isodd(direction) # flip sign of normal to make it outward pointing
```

Comment:
Did you compare if 

```julia
if direction == 1
```

is faster?

I am just curios, of course does this way neatly generalize to multiple dimensions.

Reply:
`if direction == 1` is slightly faster, will update the code. Thank you.